### PR TITLE
Add entity entity collision

### PR DESCRIPTION
**Pull request by O-of**

### CODE REVIEW

File: test/basic.test.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ function fakePlayer (pos) {
     },
     jumpTicks: 0,
     jumpQueued: false,
+    entities: {},
```
Comment (Karang) [inline test/basic.test.js:34]: You can give a list directly to simplify the interface, you don't use the key anyway

File: index.js [hunk 2]
```diff
@@ -643,13 +682,36 @@ function getStatusEffectNamesForVersion (supportFeature) {
   }
 }
 
+function getEntityBB (entity) {
+  const w = entity.width / 2
+  const pos = entity.position
+  return new AABB(-w, 0, -w, w, entity.height, w).offset(pos.x, pos.y, pos.z)
+}
+
+function getIntersectingEntities (bb, entityList) {
```
Comment (Karang) [inline index.js:691]: `entityList.filter((e) => bb.intersects(getEntityBB(e))` would be a simpler implementation ?
what is the `e.type !== 'object'` for ?

File: index.js [hunk 3]
```diff
@@ -660,6 +722,8 @@ class PlayerState {
     this.isCollidedVertically = bot.entity.isCollidedVertically
     this.jumpTicks = bot.jumpTicks
     this.jumpQueued = bot.jumpQueued
+    // Hardcoded AABB because can't access getPlayerBB()
```
Comment (Karang) [inline index.js:725]: It would be better to expose getPlayerBB()

File: index.js [hunk 4]
```diff
@@ -552,6 +585,12 @@ function Physics (mcData, world) {
     if (Math.abs(vel.y) < physics.negligeableVelocity) vel.y = 0
     if (Math.abs(vel.z) < physics.negligeableVelocity) vel.z = 0
 
+    // Assuming entity = PlayerState
+    const intersectingEntities = entity.intersectingEntities
+    for (let i = 0; i < intersectingEntities.length; i++) {
```
Comment (Karang) [inline index.js:590]: `for (const intersectingEntity of entity.intersectingEntities) {`

**Review by Karang — COMMENTED** [review]
Minor comments on the style. Otherwise it looks good. Thanks

File: index.js (hunk 2 above)
Comment (O-of) [reply index.js:691]: `e.type !== 'object'` is so we don't get pushed around by items, only entities of type `player` and `mob`
`.filter()` completely flew over my head, I'll revise it to use that instead

File: test/basic.test.js (hunk 1 above)
Comment (O-of) [reply test/basic.test.js:34]: I made `entities` an obj to more emulate a real player, but I'll change it to an array

File: index.js (hunk 3 above)
Comment (O-of) [reply index.js:725]: Should I shove `physics` into a json and move `getPlayerBB(pos)` outside of the physics class?

File: lib/features.json [hunk 5]
```diff
@@ -28,5 +28,10 @@
     "name": "effectNamesAreRegistryNames",
     "description": "Status effect names equal to their registry names",
     "versions": ["1.17"]
+  },
+  {
+    "name": "entityCollision",
+    "description": "entities can collide with each other",
+    "versions": ["1.9", "1.10", "1.11", "1.12", "1.13", "1.14", "1.15", "1.16", "1.17"]
```
Comment (Karang) [inline lib/features.json:35]: This wasn't possible in 1.8 ?

Comment (U5B) [reply lib/features.json:35]: Player collision doesn't exist in 1.8

Comment (Karang) [reply lib/features.json:35]: ok

**Review by Karang — APPROVED** [review]
LGTM (apart from conflicts that have to be resolved to merge)

**Comment by rom1504** [comment]
@O-of can you please resolve conflicts?

**Comment by Pandapip1** [comment]
I'd like it if this were something that could be disabled.

**Comment by O-of** [comment]
remaking this w/ new tests

**Pull request closed without merging**